### PR TITLE
not send a default standard to IntelliSense

### PR DESCRIPTION
**Pull request by elahehrashedi**
bug fix https://github.com/microsoft/vscode-cmake-tools/issues/1477

### CODE REVIEW

File: src/cpptools.ts [hunk 1]
```diff
@@ -117,7 +116,7 @@ export function parseCompileFlags(cptVersion: cpt.Version, args: string[], lang?
   const can_use_gnu_std = (cptVersion >= cpt.Version.v4);
   const iter = args[Symbol.iterator]();
   const extraDefinitions: string[] = [];
-  let standard: StandardVersion = (lang === 'C') ? 'c11' : 'c++17';
+  let standard: StandardVersion | undefined = undefined;
```
Comment (andreeis) [inline src/cpptools.ts:119]: remove " = undefined", declaring with " | undefined" and not initializing means initialized with undefined. The CI fails with a warning about this also.

File: src/cpptools.ts [hunk 2]
```diff
@@ -22,13 +22,12 @@ const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 const log = createLogger('cpptools');
 
 type Architecture = 'x86' | 'x64' | 'arm' | 'arm64' | undefined;
-type StandardVersion = "c89" | "c99" | "c11" | "c17" | "c++98" | "c++03" | "c++11" | "c++14" | "c++17" | "c++20"
-  | "gnu89" | "gnu99" | "gnu11" | "gnu17" | "gnu++98" | "gnu++03" | "gnu++11" | "gnu++14" | "gnu++17" | "gnu++20";
+type StandardVersion = "c89" | "c99" | "c11" | "c17" | "c++98" | "c++03" | "c++11" | "c++14" | "c++17" | "c++20" | "gnu89" | "gnu99" | "gnu11" | "gnu17" | "gnu++98" | "gnu++03" | "gnu++11" | "gnu++14" | "gnu++17" | "gnu++20";
```
Comment (bobbrow) [inline src/cpptools.ts:25]: I think the C++ and C versions were split onto two lines to make it easier to find where they start and end.  Can you change this back?

File: src/cpptools.ts [hunk 3]
```diff
@@ -22,13 +22,12 @@ const localize: nls.LocalizeFunc = nls.loadMessageBundle();
...
 
 
 export interface CompileFlagInformation {
   extraDefinitions: string[];
-  standard: StandardVersion;
+  standard: StandardVersion | undefined;
```
Comment (bobbrow) [inline src/cpptools.ts:30]: The more common way to do this when you're dealing with optional fields in an interface is:
```ts
  standard?: StandardVersion;
```

File: src/cpptools.ts (hunk 2 above)
Comment (elahehrashedi) [reply src/cpptools.ts:25]: I copied this line from cpptools API (SourceFileConfiguration interface in api.d.ts) to make sure we are not missing any item. My intention was not to remove the new line. I can keep it like this to be similar to cpptools API, or split it between C and C++.

Comment (bobbrow) [reply src/cpptools.ts:25]: I would like to keep the c++ and c versions on different lines.  [In api.ts of the cpptools API](https://github.com/microsoft/vscode-cpptools-api/blob/8ae425a623649b3f26bebad8e0917f11848506e8/api.ts#L176), you will see that they are split this way.  When api.d.ts is generated, the generator removes the newline.

**Review by andreeis — APPROVED** [review]

**Pull request merged**